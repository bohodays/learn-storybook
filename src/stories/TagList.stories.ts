import type { Meta, StoryObj } from "@storybook/react-vite";

import { fn } from "storybook/test";
import TagList from "../components/TagList";

// Docs 정보
const meta = {
  title: "Buttons/TagList", // 경로를 나타냄
  component: TagList,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
  argTypes: {
    tagList: {
      control: "object",
      desciption: "태그 리스트",
      defaultValue: ["tag1", "tag2", "tag3"],
    },
    onTagClick: { action: "clicked", description: "버튼 클릭 이벤트" },
  },
} satisfies Meta<typeof TagList>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    tagList: ["tag1", "tag2", "tag3"],
    onTagClick: () => {},
  },
};
