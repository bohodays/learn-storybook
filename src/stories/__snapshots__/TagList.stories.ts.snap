// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons/TagList Default smoke-test 1`] = `
<div class="flex gap-x-4">
  <button class="h-[33px] rounded-tag-button px-[10px] border border-white text-sm font-medium bg-white text-primary">
    tag1
  </button>
  <button class="h-[33px] rounded-tag-button px-[10px] border border-white text-sm font-medium bg-dark-opacity text-white">
    tag2
  </button>
  <button class="h-[33px] rounded-tag-button px-[10px] border border-white text-sm font-medium bg-dark-opacity text-white">
    tag3
  </button>
</div>
`;
